### PR TITLE
New reader: docs addition

### PR DESCRIPTION
**Pull request by sbesson**
Following-up on the work carried out in https://github.com/openmicroscopy/bioformats/pull/1755, this PR backports some of the discussion from the PR into the Javadocs and the Sphinx developer documentation.

- check the Javadocs for `MetadataTools.populatePixels()` describes the various signatures
- check the sections in the reader writing guide about the OME-XML metadata intiialization and the order creation for the model objects are redable

### CODE REVIEW

**Comment by hflynn** [comment]
New docs section reads fine to me. One very minor quibble for consistency - the core variable is marked as 'core' rather than using the proper mark-up in this first paragraph of this bullet point, could you add the mark-up there too, please?

**Comment by sbesson** [comment]
@hflynn: pushed with additional hyperlinks.

File: docs/sphinx/developers/reader-guide.txt [hunk 1]
```diff
@@ -74,6 +74,32 @@ Methods to override
...
+  If the reader includes metadata at the plane level, you can initialize the
+  Plane elements under the Pixels using
+  :javadoc:`MetadataTools.populatePixels(MetadataStore, FormatReader, doPlane) <loci/formats/MetadataTools.html#populatePixels(loci.formats.meta.MetadataStore, loci.formats.IFormatReader, boolean)>`::
+
+    MetadataTools.populatePixels(store, this, true);
+
+  Once the metadatastore has been initialized with the core properties,
+  additional metadata can be added to it using the setter methods. Note that
+  for each of the model components, the ``setObjectID()`` method should be
+  called before any of the ``setObjectProperty()``, e.g.::
```
Comment (rleigh-dundee) [inline docs/sphinx/developers/reader-guide.txt]: of the ``setObjectProperty()`` methods

("methods" is missing)

**Comment by rleigh-dundee** [comment]
@sbesson: Sphinx doesn't have Java domains, but you could reuse the C++ ones for Java markup, i.e.

```
:cpp:var:`core`
```

for this case

File: components/formats-api/src/loci/formats/MetadataTools.java [hunk 2]
```diff
@@ -86,17 +86,33 @@ private MetadataTools() { }
...
+   * metadata from the given reader.  If the {@code doPlane} flag is set,
+   * then the Plane elements will be populated as well.
+   *
+   * Delegates to
+   * {@link #populatePixels(MetadataStore, IFormatReader, boolean, boolean)}
+   * with {@code doImageName} set to {@code true}.
+   *
+   * @param store   The metadata store which Pixels should be populated
+   * @param r       The format reader which core metadata should be used
+   * @param doPlane Specifies whether Plane elements should be populatedZZ
```
Comment (melissalinkert) [inline components/formats-api/src/loci/formats/MetadataTools.java]: Trailing ZZ?

File: components/formats-api/src/loci/formats/MetadataTools.java [hunk 3]
```diff
@@ -86,17 +86,33 @@ private MetadataTools() { }
...
+   * Populates the Pixels element of the given metadata store, using core
+   * metadata from the given reader.  If the {@code doPlane} flag is set,
+   * then the Plane elements will be populated as well.
+   *
+   * Delegates to
+   * {@link #populatePixels(MetadataStore, IFormatReader, boolean, boolean)}
+   * with {@code doImageName} set to {@code true}.
+   *
+   * @param store   The metadata store which Pixels should be populated
+   * @param r       The format reader which core metadata should be used
```
Comment (melissalinkert) [inline components/formats-api/src/loci/formats/MetadataTools.java]: s/which/whose ?

File: components/formats-api/src/loci/formats/MetadataTools.java [hunk 4]
```diff
@@ -105,11 +121,16 @@ public static void populatePixels(MetadataStore store, IFormatReader r,
...
-   * If the 'doImageName' flag is set, then the image name will be populated
-   * as well.  By default, 'doImageName' is true.
+   * Populates the Pixels element of the given metadata store, using core
+   * metadata from the given reader.  If the {@code doPlane} flag is set,
+   * then the Plane elements will be populated as well. If the
+   * {@code doImageName} flag is set, then the image name will be populated as
+   * well.
+   *
+   * @param store       The metadata store which Pixels should be populated
+   * @param r           The format reader which core metadata should be used
```
Comment (melissalinkert) [inline components/formats-api/src/loci/formats/MetadataTools.java]: s/which/whose ?

**Comment by sbesson** [comment]
Thanks @melissalinkert, pushed.

**Pull request merged**